spotinst/commons/administration_org_programmatic_user/fields_spotinst_administration_org_programmatic_user.go

```diff
@@ -0,0 +1,312 @@
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			if v, ok := resourceData.GetOk(string(Name)); ok && v != "" {
+				orgProgUser.SetName(spotinst.String(resourceData.Get(string(Name)).(string)))
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			if v, ok := resourceData.GetOk(string(Name)); ok && v != "" {
+				orgProgUser.SetName(spotinst.String(resourceData.Get(string(Name)).(string)))
+			}
+			return nil
+		},
+		nil,
+	)
+
+	fieldsMap[Description] = commons.NewGenericField(
+		commons.AdministrationOrgProgrammaticUser,
+		Description,
+		&schema.Schema{
+			Type:     schema.TypeString,
+			Optional: true,
+			ForceNew: true,
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			var value *string = nil
+			if orgProgUser.Description != nil {
+				value = orgProgUser.Description
+			}
+			if err := resourceData.Set(string(Description), value); err != nil {
+				return fmt.Errorf(string(commons.FailureFieldReadPattern), string(Description), err)
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			if v, ok := resourceData.GetOk(string(Description)); ok && v != "" {
+				orgProgUser.SetDescription(spotinst.String(resourceData.Get(string(Description)).(string)))
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			if v, ok := resourceData.GetOk(string(Description)); ok && v != "" {
+				orgProgUser.SetDescription(spotinst.String(resourceData.Get(string(Description)).(string)))
+			}
+			return nil
+		},
+		nil,
+	)
+
+	fieldsMap[Policies] = commons.NewGenericField(
+		commons.AdministrationOrgProgrammaticUser,
+		Policies,
+		&schema.Schema{
+			Type:     schema.TypeSet,
+			Optional: true,
+			ForceNew: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					string(PolicyAccountIds): {
+						Type:     schema.TypeList,
+						Required: true,
+						Elem:     &schema.Schema{Type: schema.TypeString},
+					},
+					string(PolicyId): {
+						Type:     schema.TypeString,
+						Required: true,
```

Comment:
re-check if it's required param or optional

Reply:
Required tested thru postman